### DISCUSSION TITLE
How can I use uv_getaddrinfo_t in an on_resolved callback?

### DISCUSSION THREAD

**Discussion by nullndvoid** [libuv/libuv#3678] [Q&A] [answered]
Code attatched below, I am following the user guide example on DNS resolutions, how can I use ther resolver parameter so that my unused-parameter warnings disappear.

Thanks for any and all help,

```c

void on_resolved(uv_getaddrinfo_t* resolver, int status, struct addrinfo* res)
{
    if (status < 0) {
        fprintf(stderr, "getaddrinfo callback failed: %s\n", uv_err_name(status));
        return;
    }

    char* addr = (char*)malloc(sizeof(char) * res->ai_addrlen);
    uv_ip4_name((struct sockaddr_in*)res->ai_addr, addr, res->ai_addrlen);
    fprintf(stderr, "%s\n", addr);

    free(addr);
    uv_freeaddrinfo(res);
}

int main()
{
    uv_loop_t* loop = uv_default_loop();

    struct addrinfo hints;
    hints.ai_family = AF_UNSPEC;
    hints.ai_socktype = SOCK_STREAM;
    // works out our own IP for us.
    hints.ai_flags = AI_PASSIVE;

    // use the DNS our way.
    uv_getaddrinfo_t resolver;
    uv_getaddrinfo(loop, &resolver, on_resolved,
        NULL, "6969", &hints);

    return uv_run(loop, UV_RUN_DEFAULT);
}

```

**Comment by nullndvoid** [comment]
Correct me if I am wrong. I have figured out that `resolver->addrinfo` is equal to `res`, it is annoying that I can't omit that last function argument as it breaks the function pointer expected by uv_getaddrinfo.

**Comment by nullndvoid** [comment]
Or in other words, get rid of the warning, I have resorted to this until I work it out or someone tells me I am doing it all wrong.

```c
    if (resolver->addrinfo->ai_next != res->ai_next) {
        printf("to quell warnings, I think these should be the same.\n");
    }
```

**Comment by bnoordhuis** [comment, marked as answer]
The `uv_getaddrinfo_t*` param is the "request" in libuv parlance. People use it to carry over state through the `resolver->data` field, or by embedding it in a larger struct and then retrieving that with a container_of-like macro.

But! You don't have to use it. If you don't, just squelch the compiler warning with `(void) &resolver;`

**Reply by nullndvoid** [reply]
Thanks, that seems like a good answer
